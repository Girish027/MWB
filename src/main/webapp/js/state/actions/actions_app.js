
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
