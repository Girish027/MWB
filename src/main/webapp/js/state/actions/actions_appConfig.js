import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
