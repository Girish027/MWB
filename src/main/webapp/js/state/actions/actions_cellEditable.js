import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
