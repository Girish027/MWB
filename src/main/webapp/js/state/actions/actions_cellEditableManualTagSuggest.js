import store from 'state/configureStore';
import errorMessageUtil from 'utils/ErrorMessageUtil';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as types from './types';

let controller;
export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE,
  stateKey,
});
export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_REMOVE,
  stateKey,
});

const getSuggestedTagsRequest = ({ stateKey, projectId, intent }) => ({
  type: types.GET_SUGGESTED_TAGS_REQUEST,
  stateKey,
  projectId,
  intent,
});

const getSuggestedTagsError = ({
  stateKey, projectId, intent, error,
}) => ({
  type: types.GET_SUGGESTED_TAGS_ERROR,
  stateKey,
  projectId,
  intent,
  error,
});

const getSuggestedTagsSuccess = ({
  stateKey, projectId, intent, suggestedTags,
}) => ({
  type: types.GET_SUGGESTED_TAGS_SUCCESS,
  stateKey,
  projectId,
  intent,
  suggestedTags,
});

export const getSuggestedTags = ({ stateKey, projectId, intent }) => (dispatch, getState) => new Promise((resolve, reject) => {
  if (controller) {
    controller.abort();
  }
  controller = new AbortController();
  const signal = controller.signal;
  dispatch(getSuggestedTagsRequest({ stateKey, projectId, intent }));
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.suggestedTags, { projectId, intent, clientId });

  fetch(fetchUrl, {
    signal,
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      const suggestedTags = json && json.length ? json : [];
      const intentIndex = suggestedTags.indexOf(intent);
      if (intentIndex >= 0) {
        suggestedTags.slice(intentIndex, 1);
      }
      dispatch(getSuggestedTagsSuccess({
        stateKey, projectId, intent, suggestedTags,
      }));
      resolve([...suggestedTags]);
    })
    .catch((error) => {
      if (error.name === 'AbortError') { return; }
      dispatch(getSuggestedTagsError({
        stateKey, projectId, intent, error,
      }));
      reject(error);
    });
});

export const cursorDown = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_DOWN,
  stateKey,
});

export const cursorUp = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_UP,
  stateKey,
});
