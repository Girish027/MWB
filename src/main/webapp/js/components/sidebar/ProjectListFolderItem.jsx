import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ListItem from 'grommet/components/ListItem';
import Box from 'grommet/components/Box';
import * as actionsApp from 'state/actions/actions_app';
import * as actionsProjects from 'state/actions/actions_projects';
import isFeatureEnabled, { featureFlagDefinitions } from 'utils/FeatureFlags';
import Constants from 'constants/Constants';
import { RouteNames } from 'utils/routeHelpers';
import {
  DashedSpinner,
  SolidTriangle,
  MessageIcon,
  DropDown,
  Tooltip,
} from '@tfs/ui-components';
import MenuIcon from 'components/Icons/MenuIcon';
import IconButton from 'components/IconButton';

export default class ProjectListFolderItem extends Component {
  constructor(props, context) {
    super(props, context);

    this.onMouseEnter = this.onMouseEnter.bind(this);
    this.onMouseLeave = this.onMouseLeave.bind(this);
    this.onClickDeleteProject = this.onClickDeleteProject.bind(this);
    this.onClickPromoteProject = this.onClickPromoteProject.bind(this);
    this.onClickDemoteProject = this.onClickDemoteProject.bind(this);
    this.onClickEditProject = this.onClickEditProject.bind(this);
    this.spinnerProps = { height: '20px', width: '20px', fill: '#004C97' };
    this.props = props;

    this.state = {
      showAction: false,
      showDropDown: false,
    };
  }

  onClickPromoteOk(projectId, globalProjectName) {
    const { dispatch, globalProjectsByName } = this.props;
    const globalProjectId = globalProjectsByName[globalProjectName.toLowerCase()];
    dispatch(actionsProjects.promoteProjectById({ projectId, globalProjectId, globalProjectName }));
    dispatch(actionsProjects.updateProjectStatus(projectId, Constants.STATUS.RUNNING));
    dispatch(actionsApp.modalDialogChange(null));
  }

  onClickPromoteProject() {
    const {
      project, dispatch,
    } = this.props;

    const { id, name } = project;
    const header = `Promote '${name}' model?`;

    dispatch(actionsApp.modalDialogChange({
      header,
      dispatch,
      type: Constants.DIALOGS.PROMOTE_DIALOG,
      onClickPromote: (globalProjectName) => this.onClickPromoteOk(id, globalProjectName),
    }));
  }

  onClickDemoteOk(projectId) {
    const { dispatch } = this.props;
    dispatch(actionsProjects.demoteProjectById(projectId));
    dispatch(actionsProjects.updateProjectStatus(projectId, Constants.STATUS.RUNNING));
    dispatch(actionsApp.modalDialogChange(null));
  }

  onClickDemoteProject() {
    const {
      project, dispatch,
    } = this.props;

    const { id, name } = project;
    const header = `Demote '${name}' model?`;
    const message = 'Are you sure you want to demote this model?';

    dispatch(actionsApp.modalDialogChange({
      header,
      dispatch,
      message,
      type: Constants.DIALOGS.DELETE_DIALOG,
      ok: Constants.DEMOTE,
      onOk: () => this.onClickDemoteOk(id),
    }));
  }

  onClickDeleteOk() {
    const {
      project, selectedProjectId, dispatch, history,
    } = this.props;
    const { id, clientId } = project;
    dispatch(actionsProjects.deleteProjectById(id));
    dispatch(actionsProjects.updateProjectStatus(id, Constants.STATUS.RUNNING));
    dispatch(actionsApp.modalDialogChange(null));
    if (id === selectedProjectId) {
      dispatch(actionsApp.changeRoute(RouteNames.PROJECTS, { clientId }, history));
    }
  }

  onClickDeleteProject() {
    const {
      project, dispatch,
    } = this.props;

    const { name } = project;
    const header = `Delete '${name}' model?`;
    const message = (
      <div>
        <p>
         All your datasets and versions will be permanently deleted.
         Are you sure you want to delete this model?
        </p>
        <p>
         This cannot be reversed.
        </p>
      </div>
    );

    dispatch(actionsApp.modalDialogChange({
      header,
      dispatch,
      message,
      type: Constants.DIALOGS.DELETE_DIALOG,
      onOk: () => this.onClickDeleteOk(),
    }));
  }

  onClickEditProject() {
    const { project, dispatch, history } = this.props;
    const { id, clientId } = project;
    dispatch(actionsProjects.sidebarSelectProject({ projectId: id }));
    dispatch(actionsApp.changeRoute(RouteNames.UPDATEPROJECT, { clientId, projectId: id }, history));
  }

  onMouseEnter(event) {
    event.preventDefault();
    this.setState({
      showAction: true,
    });
  }

  onMouseLeave(event) {
    event.preventDefault();
    this.setState({
      showAction: false,
      showDropDown: false,
    });
  }

  getSpinner() {
    const { status = '' } = this.props;
    if (status === Constants.STATUS.RUNNING) {
      return (
        <DashedSpinner {...this.spinnerProps} />
      );
    }
    return null;
  }

  onDropDownSelected = (selectedValue) => {
    let result;
    switch (selectedValue) {
    case Constants.DROPDOWN_PROMOTE:
      result = this.onClickPromoteProject();
      break;
    case Constants.DROPDOWN_DEMOTE:
      result = this.onClickDemoteProject();
      break;
    case Constants.DROPDOWN_EDIT:
      result = this.onClickEditProject();
      break;
    case Constants.DROPDOWN_DELETE:
      result = this.onClickDeleteProject();
      break;
    }
    this.setState({
      showDropDown: false,
    });
    return result;
  }

  getMenuIcon(showIcon, showDemote) {
    const { userFeatureConfiguration } = this.props;
    const { names } = featureFlagDefinitions;
    const { showDropDown } = this.state;
    const dropDownList = [];
    if (showIcon) {
      dropDownList.push(Constants.DROPDOWN_PROMOTE);
      dropDownList.push(Constants.DROPDOWN_EDIT);
      if (isFeatureEnabled(names.projectDelete, userFeatureConfiguration)) {
        dropDownList.push(Constants.DROPDOWN_DELETE);
      }
    } else if (showDemote) {
      dropDownList.push(Constants.DROPDOWN_DEMOTE);
    }
    return (
      <span style={{ overflow: 'visible' }}>
        <IconButton
          onClick={() => {
            const { showDropDown: isOpen } = this.state;
            this.setState({
              showDropDown: !isOpen,
            });
          }}
          icon={MenuIcon}
          styleOverride={{
            outline: 'none',
          }}
          width={70}
          height={70}
        />
        <span onMouseLeave={() => { this.setState({ showDropDown: false }); }} style={{ overflow: 'visible' }}>
          {showDropDown
          && (
            <DropDown
              onItemSelected={(selectedValue) => { this.onDropDownSelected(selectedValue); }}
              itemList={dropDownList}
              styleOverride={{
                menuList: {
                  top: 10,
                  right: 55,
                },
              }}
            />
          )
          }
        </span>
      </span>
    );
  }

  truncateCount(number) {
    if (number.toString().length > 8) {
      return `${number.toString().substring(0, 7)}...`;
    }
    return number;
  }

  render() {
    const {
      project = {}, onClick, status = '',
    } = this.props;
    const { showAction } = this.state;
    const {
      id,
      name,
      type,
    } = project;
    const showIcon = (type === Constants.PROJECT_TYPE.NODE.NAME && showAction && status !== Constants.STATUS.RUNNING);
    const showDemote = (showAction && status !== Constants.STATUS.RUNNING);
    const versionNumber = 65;
    const volumeCount = 8000;
    const escalationCount = 2850;
    const volumePercentange = 1.6;
    const escalationPercentange = 1.8;
    const redEscalationTheme = false;

    return (
      <ListItem
        justify="between"
        key={id}
        onClick={onClick}
        selected={this.props.selected}
        className="ProjectListFolderItem"
        onMouseEnter={this.onMouseEnter}
        onMouseLeave={this.onMouseLeave}
      >
        <Box direction="row" flex>
          <Box
            direction="row"
            align="center"
            className="ProjectListFolderIcon"
            style={redEscalationTheme ? { backgroundColor: '#e03c31' } : { backgroundColor: '#d7d8d9' }}
          >
            <MessageIcon stroke={redEscalationTheme ? '#f8e8e7' : '#7f7f7f'} />
          </Box>
          <Box
            flex
            className="ProjectListFolderItemName"
          >
            <Box direction="row">
              <Box>
                <span className="project-list-menu-icon">
                  { showAction && this.getMenuIcon(showIcon, showDemote)}
                  { this.getSpinner() }
                </span>
              </Box>
              <div className="project-list-item-name">
                {name}
              </div>
            </Box>
            <div className="project-list-item-meta">
              {Constants.VERSION(versionNumber)}
            </div>
            <div className="project-list-item-duration">
              {'LAST TWO WEEKS'}
            </div>
            <hr />
            <Box direction="row">
              <Box style={{ width: '130px' }}>
                <div>
                  <span
                    className="project-list-volume-count"
                  >
                    <Tooltip
                      content={volumeCount}
                      direction="right"
                      enabled={volumeCount.length > 8}
                    >
                      <span>
                        {this.truncateCount(volumeCount)}
                      </span>
                    </Tooltip>
                  </span>
                  <span className="grey-theme-volume">
                    <SolidTriangle
                      fill="#6E7887"
                      direction="down"
                    />
                    {Constants.PERCENTAGE(volumePercentange)}
                  </span>
                  <div className="project-list-volume">
                    {Constants.VOLUME}
                  </div>
                </div>
              </Box>
              <Box style={{ width: '130px' }}>
                <div>
                  <span
                    className="project-list-escalation-count"
                  >
                    <Tooltip
                      content={escalationCount}
                      direction="right"
                      enabled={escalationCount.length > 8}
                    >
                      <span>
                        {this.truncateCount(escalationCount)}
                      </span>
                    </Tooltip>
                  </span>
                  <span className={redEscalationTheme ? 'red-theme-escalation' : 'grey-theme-escalation'}>
                    <SolidTriangle
                      fill={redEscalationTheme ? '#e03c31' : '#6E7887'}
                      direction="up"
                    />
                    {Constants.PERCENTAGE(escalationPercentange)}
                  </span>
                  <div className="project-list-escalation">
                    {Constants.ESCALATION}
                  </div>
                </div>
              </Box>
            </Box>
          </Box>
        </Box>
      </ListItem>
    );
  }
}

ProjectListFolderItem.propTypes = {
  index: PropTypes.number,
  project: PropTypes.object.isRequired,
  status: PropTypes.string,
  onClick: PropTypes.func,
  selected: PropTypes.bool,
  dispatch: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
};
