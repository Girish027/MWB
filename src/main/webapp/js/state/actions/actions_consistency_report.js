import errorMessageUtil from 'utils/ErrorMessageUtil';
import store from 'state/configureStore';

import getUrl, { pathKey } from 'utils/apiUrls';
import * as appActions from './actions_app';
import * as types from './types';

export const reset = () => ({
  type: types.CONSISTENCY_REPORT_RESET,
});

export const setProjectId = projectId => ({
  type: types.CONSISTENCY_REPORT_SET_PROJECT_ID,
  projectId,
});

export const setProject = ({ projectId, project }) => ({
  type: types.CONSISTENCY_REPORT_SET_PROJECT,
  projectId,
  project,
});

export const setDatasets = ({ projectId, datasets }) => ({
  type: types.CONSISTENCY_REPORT_SET_DATASETS,
  projectId,
  datasets,
});


export const setFilter = ({ datasets, onlyConflicts }) => ({
  type: types.CONSISTENCY_REPORT_SET_FILTER,
  datasets,
  onlyConflicts,
});

const requestSearchSuccess = ({ results, startIndex }) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_SUCCESS,
  results,
  startIndex,
});

const requestSearchRequest = ({
  promise, projectId, query, filter, sort, startIndex, limit,
}) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_REQUEST,
  promise,
  projectId,
  query,
  filter,
  sort,
  startIndex,
  limit,
});

const requestSearchError = ({ error }) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_ERROR,
  error,
});

export const setPagerSettings = ({ startIndex, limit, showControls }) => ({
  type: types.CONSISTENCY_REPORT_SET_PAGER_SETTINGS,
  startIndex,
  limit,
  showControls,
});

export const requestSearch = ({
  projectId, query, filter, sort, startIndex, limit, delay,
}, isNested) => (dispatch, getState) => {
  const promise = new Promise((resolve, reject) => {
    const state = getState();
    const { userId, csrfToken } = state.app;
    const clientId = state.header.client.id;

    const fetchUrl = getUrl(pathKey.consistency, {
      projectId,
      startIndex,
      limit,
      sortProperty: sort.property,
      sortDirection: sort.direction,
      clientId,
    });

    const filterBodyObj = {};
    filterBodyObj.onlyTagged = true;
    filterBodyObj.onlyConflicts = filter.onlyConflicts;
    if (filter.datasets && filter.datasets.length) {
      filterBodyObj.datasetIds = filter.datasets;
    }

    setTimeout(() => {
      fetch(fetchUrl, {
        credentials: 'same-origin',
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'X-CSRF-TOKEN': csrfToken,
        },
        body: JSON.stringify({
          configId: '0',
          filter: filterBodyObj,
        }),
      })
        .then(errorMessageUtil.handleErrors)
        .then(response => response.json())
        .then((results) => {
          const searchResults = results.transcriptionList || [];

          searchResults.forEach((res, index) => {
            res._rowIndex = index;
          });

          if (!searchResults.length && startIndex > 0) {
            const total = results.total;
            const maxPage = Math.max(Math.floor(total / limit) + (total % limit ? 1 : 0), 1);
            const newStartIndex = (maxPage - 1) * limit;
            dispatch(requestSearch({
              projectId,
              query,
              filter,
              sort,
              startIndex: newStartIndex,
              limit,
            }, true))
              .then((results) => {
                dispatch(requestSearchSuccess({ results, startIndex: newStartIndex }));
                resolve(results);
              })
              .catch((error) => {
                reject(error);
              });
          } else {
            if (!isNested) {
              dispatch(requestSearchSuccess({ results, startIndex }));
            }
            resolve(results);
          }
        })
        .catch((error) => {
          dispatch(requestSearchError({ error }));
          errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
          reject(error);
        });
    }, delay || 0);
  });

  if (!isNested) {
    dispatch(requestSearchRequest({
      promise, projectId, query, filter, sort, startIndex, limit,
    }));
  }

  return promise;
};

export const refresh = ({ delay }) => ({
  type: types.CONSISTENCY_REPORT_REFRESH,
  delay,
});
