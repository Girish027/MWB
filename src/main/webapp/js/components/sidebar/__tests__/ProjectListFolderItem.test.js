import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ProjectListFolderItem from 'components/sidebar/ProjectListFolderItem';
import * as actionsApp from 'state/actions/actions_app';

describe('<ProjectListFolderItem />', () => {
  const initialState = {
    showAction: false,
    showDropDown: false,
  };

  const testProps = {
    project: {
      clientId: '151',
      id: '2358',
      name: 'Root_Intent',
      type: 'GLOBAL',
    },
    index: 0,
    dispatch: jest.fn(),
  };

  let wrapper;

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (props) => shallow(<ProjectListFolderItem
    {...props}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = shallow(<ProjectListFolderItem
        {...testProps}
      />);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    beforeEach(() => {
      wrapper = shallow(<ProjectListFolderItem
        {...testProps}
      />);
      wrapper.setState({
        ...initialState,
      });
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    test('should match with basic props', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      actionsApp.changeRoute = jest.fn(() => 'called changeRoute');
      actionsApp.modalDialogChange = jest.fn(() => 'called modalDialogChange');
    });
    test('should call onDropDownSelected with value Promote', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickPromoteProject = jest.fn();
      wrapper.instance().onDropDownSelected('Promote');
      expect(wrapper.instance().onClickPromoteProject).toHaveBeenCalled();
      expect(wrapper.state().showDropDown).toBe(false);
    });

    test('should call onDropDownSelected with value Demote', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickDemoteProject = jest.fn();
      wrapper.instance().onDropDownSelected('Demote');
      expect(wrapper.instance().onClickDemoteProject).toHaveBeenCalled();
    });

    test('should call onDropDownSelected with value Delete', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickDeleteProject = jest.fn();
      wrapper.instance().onDropDownSelected('Delete');
      expect(wrapper.instance().onClickDeleteProject).toHaveBeenCalled();
    });

    test('should call onDropDownSelected with value Edit', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickEditProject = jest.fn();
      wrapper.instance().onDropDownSelected('Edit');
      expect(wrapper.instance().onClickEditProject).toHaveBeenCalled();
    });

    test('should call onClickPromoteProject', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickPromoteProject();
      expect(actionsApp.modalDialogChange).toHaveBeenCalled();
      expect(testProps.dispatch).toHaveBeenCalledWith('called modalDialogChange');
    });

    test('should call onClickDeleteProject', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickDeleteProject();
      expect(actionsApp.modalDialogChange).toHaveBeenCalled();
      expect(testProps.dispatch).toHaveBeenCalledWith('called modalDialogChange');
    });

    test('should call onClickDemoteProject', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickDemoteProject();
      expect(actionsApp.modalDialogChange).toHaveBeenCalled();
      expect(testProps.dispatch).toHaveBeenCalledWith('called modalDialogChange');
    });

    test('should call onClickEditProject', () => {
      wrapper = getShallowWrapper(testProps);
      wrapper.instance().onClickEditProject();
      expect(actionsApp.changeRoute).toHaveBeenCalled();
      expect(testProps.dispatch).toHaveBeenCalledWith('called changeRoute');
    });
  });
});
