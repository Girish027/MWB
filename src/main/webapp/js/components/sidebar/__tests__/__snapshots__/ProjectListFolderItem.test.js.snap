// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectListFolderItem /> Snapshots should match with basic props 1`] = `
<ListItem
  align="center"
  className="ProjectListFolderItem"
  direction="row"
  justify="between"
  key="2358"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  pad={
    Object {
      "horizontal": "medium",
      "vertical": "small",
    }
  }
  role="listitem"
  separator="bottom"
>
  <Box
    announce={false}
    direction="row"
    flex={true}
    focusable={true}
    pad="none"
    responsive={true}
    tag="div"
  >
    <Box
      align="center"
      announce={false}
      className="ProjectListFolderIcon"
      direction="row"
      focusable={true}
      pad="none"
      responsive={true}
      style={
        Object {
          "backgroundColor": "#d7d8d9",
        }
      }
      tag="div"
    >
      <MessageIcon
        color="none"
        height="19.25px"
        stroke="#7f7f7f"
        width="19.25px"
      />
    </Box>
    <Box
      announce={false}
      className="ProjectListFolderItemName"
      direction="column"
      flex={true}
      focusable={true}
      pad="none"
      responsive={true}
      tag="div"
    >
      <Box
        announce={false}
        direction="row"
        focusable={true}
        pad="none"
        responsive={true}
        tag="div"
      >
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad="none"
          responsive={true}
          tag="div"
        >
          <span
            className="project-list-menu-icon"
          />
        </Box>
        <div
          className="project-list-item-name"
        >
          Root_Intent
        </div>
      </Box>
      <div
        className="project-list-item-meta"
      >
        Version 65
      </div>
      <div
        className="project-list-item-duration"
      >
        LAST TWO WEEKS
      </div>
      <hr />
      <Box
        announce={false}
        direction="row"
        focusable={true}
        pad="none"
        responsive={true}
        tag="div"
      >
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad="none"
          responsive={true}
          style={
            Object {
              "width": "130px",
            }
          }
          tag="div"
        >
          <div>
            <span
              className="project-list-volume-count"
            >
              <withTheme(Tooltip)
                content={8000}
                direction="right"
                enabled={false}
              >
                <span>
                  8000
                </span>
              </withTheme(Tooltip)>
            </span>
            <span
              className="grey-theme-volume"
            >
              <SolidTriangle
                direction="down"
                fill="#6E7887"
                height="6px"
                width="11px"
              />
              1.6%
            </span>
            <div
              className="project-list-volume"
            >
              Volume
            </div>
          </div>
        </Box>
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad="none"
          responsive={true}
          style={
            Object {
              "width": "130px",
            }
          }
          tag="div"
        >
          <div>
            <span
              className="project-list-escalation-count"
            >
              <withTheme(Tooltip)
                content={2850}
                direction="right"
                enabled={false}
              >
                <span>
                  2850
                </span>
              </withTheme(Tooltip)>
            </span>
            <span
              className="grey-theme-escalation"
            >
              <SolidTriangle
                direction="up"
                fill="#6E7887"
                height="6px"
                width="11px"
              />
              1.8%
            </span>
            <div
              className="project-list-escalation"
            >
              Escalation
            </div>
          </div>
        </Box>
      </Box>
    </Box>
  </Box>
</ListItem>
`;
